app/controllers/api/host_aggregates_controller.rb

```diff
@@ -26,37 +20,25 @@ def delete_resource_main_action(_type, host_aggregate, _data = {})
     end
 
     def add_host_resource(type, id, data = {})
-      raise BadRequestError, "Must specify an id" if id.nil?
-      raise BadRequestError, "Must specify a host_id" if data["host_id"].nil?
-
-      host_aggregate = resource_search(id, type)
-      raise BadRequestError, host_aggregate.unsupported_reason(:add_host) unless host_aggregate.supports?(:add_host)
-
-      new_host = resource_search(data["host_id"], :hosts)
-      task_id = host_aggregate.add_host_queue(current_user.userid, new_host)
-      action_result(true, "Adding #{host_aggregate_ident(host_aggregate)}", :task_id => task_id)
-    rescue => err
-      action_result(false, err.to_s)
+      api_resource(type, id, "Adding Host to", :supports => :add_host) do |host_aggregate|
+        new_host = search_host(data["host_id"])
+        {:task_id => host_aggregate.add_host_queue(User.current_userid, new_host)}
+      end
     end
 
     def remove_host_resource(type, id, data = {})
-      raise BadRequestError, "Must specify an id" if id.nil?
-      raise BadRequestError, "Must specify a host_id" if data["host_id"].nil?
-
-      host_aggregate = resource_search(id, type)
-      raise BadRequestError, host_aggregate.unsupported_reason(:remove_host) unless host_aggregate.supports?(:remove_host)
-
-      new_host = resource_search(data["host_id"], :hosts)
-      task_id = host_aggregate.remove_host_queue(current_user.userid, new_host)
-      action_result(true, "Removing #{host_aggregate_ident(host_aggregate)}", :task_id => task_id)
-    rescue => err
-      action_result(false, err.to_s)
+      api_resource(type, id, "Removing Host from", :supports => :remove_host) do |host_aggregate|
+        new_host = search_host(data["host_id"])
+        {:task_id => host_aggregate.remove_host_queue(User.current_userid, new_host)}
+      end
     end
 
     private
 
-    def host_aggregate_ident(host_aggregate)
-      "Host Aggregate id:#{host_aggregate.id} name: '#{host_aggregate.name}'"
+    def search_host(host_id)
+        raise BadRequestError, "Must specify a host_id" if data["host_id"].nil?
```

Comment:
Spoke too soon before CI finished :laughing: should be 
```suggestion
        raise BadRequestError, "Must specify a host_id" if host_id.nil?
```

Reply:
that "parameter is not being used" sure catches a lot of errors for me.